### PR TITLE
#161290964 Build local login feature

### PR DESCRIPTION
**Pull request by tersoo-atsen**
#### What does this PR do?

Have user be able to login to their Author's Haven accounts

#### Description of Task to be completed?

- add tests
- add login action creators
- add welcome page component

#### How should this be manually tested?

- run `npm start` to spin up the server
- open localhost:3000 to view the landing page
- use redux dev tool to observe the state and state history
- navigate to the login page and login with a valid user email
- go to your inbox look for the email from Author's Haven and click on the link
- you should be directed to the welcome page and logged in now

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?

[#161290964](https://www.pivotaltracker.com/story/show/161290964)

#### Screenshots (if appropriate)

screen shot 2018-12-19 at 3 07 28 pm

screen shot 2018-12-19 at 3 08 28 pm

screen shot 2018-12-16 at 12 42 26 pm

screen shot 2018-12-19 at 3 40 13 pm

#### Questions:

N/A

[Review App](https://valinor-ah-frontend-stag-pr-31.herokuapp.com/login)

### CODE REVIEW

**Review by sulenchy — CHANGES_REQUESTED** [review]
By default, the landing page displays an error from google. Below is the  snapshot of the page
screen shot 2018-12-20 at 10 13 22 am

I think it is a bug

File: src/actions/loginActions.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  .then((jsonResponse) => {
+    if (jsonResponse.errors) {
+      dispatch(globalFailure(jsonResponse.errors));
+      toastr.error('There was an error please try again');
+    } else {
+      toastr.success('Email sent successfully! Check your email.');
+    }
+  });
+
+export const logUserIn = tokenQueryString => dispatch => fetch(`${process.env.API_BASE_URL}/users/login${tokenQueryString}`)
```
Comment (andela-moe) [inline src/actions/loginActions.js]: I believe this line is longer than the agreed max length for our codebase. What can we do to fix this?

File: src/actions/loginActions.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { toastr } from 'react-redux-toastr';
+
+import { globalLoading, globalFailure, globalLoggedIn } from './globalActions';
+import requestOptions from '../utils/requestOptions';
+
+export const sendEmailLink = email => dispatch => fetch(`${process.env.API_BASE_URL}/users/login`, requestOptions({ email }, 'POST', null))
```
Comment (andela-moe) [inline src/actions/loginActions.js]: This is a violation of the max length rule. Please resolve.

**Review by andela-moe — COMMENTED** [review]
In your App.js file on line 21 you happen to be using componentWillMount() which at this point has been deprecated and moreover if you would want to check the status of the user after the component (in this cace the App) is rendered then you would consider using ComponentDIdMount versus what you currently have.

This resource should provide more insight -> https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/

File: src/actions/loginActions.js (hunk 1 above)
Comment (tersoo-atsen) [reply src/actions/loginActions.js]: I will make the line shorter.

File: src/actions/loginActions.js (hunk 2 above)
Comment (tersoo-atsen) [reply src/actions/loginActions.js]: Alright.

**Comment by tersoo-atsen** [comment]
@sulenchy this problem has been resolved. The problem was my review app had not been added to the Google developer console.

**Pull request merged**